### DISCUSSION TITLE
A way to get the Prometheus config (like `/api/v1/status/config`)

### DISCUSSION THREAD

**Discussion by tanguynicolas** [4n4nd/prometheus-api-client-python#290] [Q&A]
Hi,

Is there a way to get the Prometheus configuration? Like when going to the `/api/v1/status/config` endpoint. Or maybe a way to get the list of targets, like `/api/v1/targets` endpoint.

Didn't found in the documentation. I need this for a particular usecase.

Something like `prom.config()` or `prom.targets()`, as can be done to retrieve the list of metrics with `prom.all_metrics()`.

I'm doing this for now:

```python
full_config = requests.get(f"{prom.url}/api/v1/status/config").json()
full_config_yml = full_config['data']['yaml']
```

Thanks

**Comment by 4n4nd** [comment]
@tanguynicolas PrometheusConnect cannot do this atm. But I think this would be a great addition to the package. Would you want to work on adding it?

**Reply by tanguynicolas** [reply]
I'm just a simple system administrator. I'm afraid I can't do it. I contribute to Terraform modules because that's what I'm good at, but contributing to a Python package is something else.

Do you have a PR for a similar modification that I could use as an example?
